### PR TITLE
feat: data layer API call for ORA block data

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add a new endpoint `{{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/get_block_info` to get ORA block info.

JIRA: [AU-1381](https://2u-internal.atlassian.net/browse/AU-1381)

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

- [x] TODO ADD TESTS

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/xblock/data_layer/serializers.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    child = CharField()
+
+
+class TextResponseConfigSerializer(Serializer):
+    enabled = SerializerMethodField()
+    required = SerializerMethodField()
+    editor_type = CharField(source="text_response_editor")
+    allow_latex_preview = BooleanField(source="allow_latex")
+
+    def get_enabled(self, block):
```
Comment (jansenk) [inline openassessment/xblock/data_layer/serializers.py:22]: I wish there was a simple way to remove this semi-duplicated code across the few serializers but i think doing that would complicate things more than they need to be

File: openassessment/xblock/data_layer/serializers.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        """Get the assessment step config for a given step_name"""
+        for step in instance.rubric_assessments:
+            if step["name"] == step_name:
+                return step
+        return None
+
+    def get_training_step(self, instance):
+        """Get the training step configuration"""
+        training_step = self._get_step(instance, "student-training")
+        return TrainingSettingsSerializer(training_step).data or {}
```
Comment (jansenk) [inline openassessment/xblock/data_layer/serializers.py:149]: I wonder if it would be easy to do this as a custom Field type to remove the duplication here

Comment (nsprenkle) [reply openassessment/xblock/data_layer/serializers.py]: @jansenk , how's this new version look? https://github.com/openedx/edx-ora2/pull/2002/commits/e6c6de221dcc11832b3deab2e6b6221cacae6fa2

File: openassessment/xblock/data_layer/serializers.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    criteria = RubricCriterionSerializer(
+        many=True, source="rubric_criteria_with_labels"
+    )
+
+
+class PeerSettingsSerializer(Serializer):
+    required = BooleanField(default=True)
+
+    start = DateTimeField()
+    due = DateTimeField()
```
Comment (leangseu-edx) [inline openassessment/xblock/data_layer/serializers.py:112]:
```suggestion
    startDate = DateTimeField()
    endDate = DateTimeField()
```

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

**Review by jansenk — APPROVED** [review]
Good job, nice tests

**Comment by UsamaSadiq** [comment]
When attempting to upgrade `edx-ora2` package to the new version `edx-ora2==5.2.2` (released by arbi-bom to resolve our blocker in edx-platform due to simple-history package) the edx-platform tests are failing with the new package version. See the details in github action:  https://github.com/openedx/edx-platform/actions/runs/5797385497/job/15713532692?pr=32939#step:7:4578

**Comment by leangseu-edx** [comment]
@nsprenkle maybe add `__init__.py` to the folder? It doesn't seems `data_layer` folder get packaged.
